enigma-js/src/Enigma.js

```diff
@@ -97,15 +97,19 @@ export default class Enigma {
       // TODO: never larger that 53-bit?
       const nonce = parseInt(await this.enigmaContract.methods.getUserTaskDeployments(sender).call());
       const scAddr = isContractDeploymentTask ? utils.generateScAddr(sender, nonce) : scAddrOrPreCode;
-      const preCode = isContractDeploymentTask ? scAddrOrPreCode : '';
-
-      let preCodeArray = [];
-      for (let n = 0; n < preCode.length; n += 2) {
-        preCodeArray.push(parseInt(preCode.substr(n, 2), 16));
+      let preCode;
+      if (isContractDeploymentTask) {
+        if (Buffer.isBuffer(scAddrOrPreCode)) {
```

Comment:
Shouldn't the integration deploy scenarios be updated as well, as today the pre-code is [passed](https://github.com/enigmampc/enigma-contract/blob/develop/enigma-js/test/integrationTests/template.02_deploy_calculator.js#L54) as a hexadecimal string?

Reply:
Yes, for sure! I was waiting for you to test/implement the base64 decoding on the p2p side first and modify the integration tests later when they can be tested end to end. My rationale is that maybe you find another implementation, or want to make some other changes,  so wanted to confirm first that the proposed implementation works

Reply:
Great, thanks for the clarification